<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>metabolonIO • metabolonIO</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="metabolonIO">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">metabolonIO</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/metabolonIO.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/edo98811/metabolon_summarized_experiment/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>metabolonIO</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/edo98811/metabolon_summarized_experiment/blob/main/vignettes/metabolonIO.Rmd" class="external-link"><code>vignettes/metabolonIO.Rmd</code></a></small>
      <div class="d-none name"><code>metabolonIO.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="setup">setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/edo98811/metabolon_summarized_experiment" class="external-link">"metabolonIO"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by</span></span>
<span><span class="co">#&gt; 'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"macrophage"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">"SummarizedExperiment"</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">"limma"</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">"edgeR"</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/thelovelab/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="path-to-example-client-data-table-">Path to example client data table.<a class="anchor" aria-label="anchor" href="#path-to-example-client-data-table-"></a>
</h2>
<p>This is data saved in the package now, it will be needed later. The
cdt path is th e path to the Client data table, in this case we took the
model of a client data table and filled it up with example data</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cdt_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_data_tables.XLSX"</span>, package <span class="op">=</span> <span class="st">"metabolonIO"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">15</span><span class="op">)</span> <span class="co"># per togliere tutto quello dopo il punto</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="setting-up-example-data-">Setting up example data.<a class="anchor" aria-label="anchor" href="#setting-up-example-data-"></a>
</h2>
<p>First step is setting up a differential expression analysis using
DESeq2 and limma on the macrophage data included in the macrophage
package as example, this data will be used to show how to export it to
metabolon.</p>
<details><summary>
Setting up the data
</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">gse</span>, design <span class="op">=</span> <span class="op">~</span><span class="va">condition</span><span class="op">)</span></span>
<span><span class="co">#&gt; using counts and average transcript lengths from tximeta</span></span>
<span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">6</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="va">dds</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="co">#&gt; estimating size factors</span></span>
<span><span class="co">#&gt; using 'avgTxLength' from assays(dds), correcting for library size</span></span>
<span><span class="co">#&gt; estimating dispersions</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; fitting model and testing</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span><span class="op">[</span>, <span class="st">"condition_name"</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">condition</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">condition</span><span class="op">)</span></span>
<span></span>
<span><span class="va">contrast.matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html" class="external-link">makeContrasts</a></span><span class="op">(</span></span>
<span>  IFNg_vs_naive <span class="op">=</span> <span class="va">IFNg</span> <span class="op">-</span> <span class="va">naive</span>,</span>
<span>  levels <span class="op">=</span> <span class="va">design</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">d0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">d0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">d0</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cutoff</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">drop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">d0</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">cutoff</span><span class="op">)</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="va">d0</span><span class="op">[</span><span class="op">-</span><span class="va">drop</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">matrix_to_fit_model_to</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">voom</a></span><span class="op">(</span><span class="va">d</span>, <span class="va">design</span>, plot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="co"># matrix_to_fit_model_to &lt;- assay(se) # Use the assay data from the SummarizedExperiment object</span></span>
<span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">matrix_to_fit_model_to</span>, <span class="va">design</span><span class="op">)</span> <span class="co">#</span></span>
<span><span class="va">fit2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html" class="external-link">contrasts.fit</a></span><span class="op">(</span><span class="va">fit</span>, <span class="va">contrast.matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View(as.data.frame(colData(se)))</span></span>
<span><span class="va">fit2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit2</span><span class="op">)</span> </span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_macrophage_IFNg_vs_naive_dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span>,</span>
<span>  contrast <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"IFNg"</span>, <span class="st">"naive"</span><span class="op">)</span>,</span>
<span>  lfcThreshold <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res_macrophage_IFNg_vs_naive_dds</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; out of 17806 with nonzero total read count</span></span>
<span><span class="co">#&gt; adjusted p-value &lt; 0.05</span></span>
<span><span class="co">#&gt; LFC &gt; 1.00 (up)    : 416, 2.3%</span></span>
<span><span class="co">#&gt; LFC &lt; -1.00 (down) : 141, 0.79%</span></span>
<span><span class="co">#&gt; outliers [1]       : 95, 0.53%</span></span>
<span><span class="co">#&gt; low counts [2]     : 0, 0%</span></span>
<span><span class="co">#&gt; (mean count &lt; 3)</span></span>
<span><span class="co">#&gt; [1] see 'cooksCutoff' argument of ?results</span></span>
<span><span class="co">#&gt; [2] see 'independentFiltering' argument of ?results</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_macrophage_IFNg_vs_naive_limma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit2</span>, coef <span class="op">=</span> <span class="st">"IFNg_vs_naive"</span>, adjust <span class="op">=</span> <span class="st">"fdr"</span>, number <span class="op">=</span> <span class="cn">Inf</span>, confint <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</details>
</div>
<div class="section level2">
<h2 id="package-usage">Package usage<a class="anchor" aria-label="anchor" href="#package-usage"></a>
</h2>
<div class="section level3">
<h3 id="converting-summarizedexperiment-or-de-results-to-metabolon-format">Converting SummarizedExperiment or DE results to Metabolon
format<a class="anchor" aria-label="anchor" href="#converting-summarizedexperiment-or-de-results-to-metabolon-format"></a>
</h3>
<p>The requirement for this package to work is that the
<code>CLIENT_SAMPLE_ID</code> is the same as the colnames of the
summarized experiment.</p>
<p>You can call these two functions to convert your data to a format
that can be uploaded to the Metabolon platform in the multiomics
section. It automatically formats the different expression results to
the format requested by Metabolon.</p>
<p>It works with DESeq2 and limma results, you can also provide a custom
data frame with the required columns. The <code>cdt</code> argument is
the path to the client data table file that you want to use to get the
sample metadata.</p>
<p>In addition to this you can also convert the assay data of a
summarized experiment object to the format requested by Metabolon using
the function <code>se_to_metabolon</code>.</p>
<p>Important note: for metabolon to be able to run the pathway analysis
the feature annotation must be in the format requested by them. For
transcriptomics data the recommended format is ENSEMBL gene ids, for
proteomics data the recommended format is UNIPROT ids.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">table1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/se_to_metabolon.html">se_to_metabolon</a></span><span class="op">(</span><span class="va">gse</span>,</span>
<span>  cdt <span class="op">=</span> <span class="va">cdt_path</span>,</span>
<span>  save_file <span class="op">=</span> <span class="cn">F</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">table1</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                PARENT_SAMPLE_NAME ENSG00000000003 ENSG00000000005</span></span>
<span><span class="co">#&gt; SAMEA103885102             TEST-1          83.818               0</span></span>
<span><span class="co">#&gt; SAMEA103885347             TEST-2         144.156               0</span></span>
<span><span class="co">#&gt; SAMEA103885043             TEST-3          71.239               0</span></span>
<span><span class="co">#&gt; SAMEA103885392             TEST-4         130.372               0</span></span>
<span><span class="co">#&gt; SAMEA103885182             TEST-5         241.527               0</span></span>
<span><span class="co">#&gt; SAMEA103885136             TEST-6         140.124               0</span></span>
<span><span class="co">#&gt;                ENSG00000000419 ENSG00000000457 ENSG00000000460 ENSG00000000938</span></span>
<span><span class="co">#&gt; SAMEA103885102        1462.001         616.732         913.913         210.999</span></span>
<span><span class="co">#&gt; SAMEA103885347        1380.000        1058.364         330.360         273.001</span></span>
<span><span class="co">#&gt; SAMEA103885043        1117.000         436.315         292.047         635.000</span></span>
<span><span class="co">#&gt; SAMEA103885392         961.000         703.578         119.337        1210.000</span></span>
<span><span class="co">#&gt; SAMEA103885182        1399.000         580.188         492.361         528.000</span></span>
<span><span class="co">#&gt; SAMEA103885136        1276.000        1036.716         173.410         669.000</span></span>
<span><span class="co">#&gt;                ENSG00000000971 ENSG00000001036 ENSG00000001084</span></span>
<span><span class="co">#&gt; SAMEA103885102         508.000        7715.294        6363.152</span></span>
<span><span class="co">#&gt; SAMEA103885347        8355.997        6376.500       15601.710</span></span>
<span><span class="co">#&gt; SAMEA103885043         657.999        6231.000        2356.371</span></span>
<span><span class="co">#&gt; SAMEA103885392        6407.001        5307.074        8890.247</span></span>
<span><span class="co">#&gt; SAMEA103885182         140.999        5099.999        5275.775</span></span>
<span><span class="co">#&gt; SAMEA103885136        6670.997        4314.000       14894.069</span></span>
<span></span>
<span><span class="va">res_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/results_to_metabolon.html">results_to_metabolon</a></span><span class="op">(</span><span class="va">res_macrophage_IFNg_vs_naive_dds</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"DESeqResults"</span>,</span>
<span>  omics <span class="op">=</span> <span class="st">"transcriptomics"</span>,</span>
<span>  custom_colnames <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  save_file <span class="op">=</span> <span class="cn">F</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">table1</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                PARENT_SAMPLE_NAME ENSG00000000003 ENSG00000000005</span></span>
<span><span class="co">#&gt; SAMEA103885102             TEST-1          83.818               0</span></span>
<span><span class="co">#&gt; SAMEA103885347             TEST-2         144.156               0</span></span>
<span><span class="co">#&gt; SAMEA103885043             TEST-3          71.239               0</span></span>
<span><span class="co">#&gt; SAMEA103885392             TEST-4         130.372               0</span></span>
<span><span class="co">#&gt; SAMEA103885182             TEST-5         241.527               0</span></span>
<span><span class="co">#&gt; SAMEA103885136             TEST-6         140.124               0</span></span>
<span><span class="co">#&gt;                ENSG00000000419 ENSG00000000457 ENSG00000000460 ENSG00000000938</span></span>
<span><span class="co">#&gt; SAMEA103885102        1462.001         616.732         913.913         210.999</span></span>
<span><span class="co">#&gt; SAMEA103885347        1380.000        1058.364         330.360         273.001</span></span>
<span><span class="co">#&gt; SAMEA103885043        1117.000         436.315         292.047         635.000</span></span>
<span><span class="co">#&gt; SAMEA103885392         961.000         703.578         119.337        1210.000</span></span>
<span><span class="co">#&gt; SAMEA103885182        1399.000         580.188         492.361         528.000</span></span>
<span><span class="co">#&gt; SAMEA103885136        1276.000        1036.716         173.410         669.000</span></span>
<span><span class="co">#&gt;                ENSG00000000971 ENSG00000001036 ENSG00000001084</span></span>
<span><span class="co">#&gt; SAMEA103885102         508.000        7715.294        6363.152</span></span>
<span><span class="co">#&gt; SAMEA103885347        8355.997        6376.500       15601.710</span></span>
<span><span class="co">#&gt; SAMEA103885043         657.999        6231.000        2356.371</span></span>
<span><span class="co">#&gt; SAMEA103885392        6407.001        5307.074        8890.247</span></span>
<span><span class="co">#&gt; SAMEA103885182         140.999        5099.999        5275.775</span></span>
<span><span class="co">#&gt; SAMEA103885136        6670.997        4314.000       14894.069</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="converting-de-results-to-metabolon-format">Converting DE results to Metabolon format<a class="anchor" aria-label="anchor" href="#converting-de-results-to-metabolon-format"></a>
</h3>
<p>The package works also with limma results using topTable format:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/results_to_metabolon.html">results_to_metabolon</a></span><span class="op">(</span><span class="va">res_macrophage_IFNg_vs_naive_limma</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"topTable"</span>,</span>
<span>  omics <span class="op">=</span> <span class="st">"transcriptomics"</span>,</span>
<span>  custom_colnames <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  save_file <span class="op">=</span> <span class="cn">F</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">table1</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                PARENT_SAMPLE_NAME ENSG00000000003 ENSG00000000005</span></span>
<span><span class="co">#&gt; SAMEA103885102             TEST-1          83.818               0</span></span>
<span><span class="co">#&gt; SAMEA103885347             TEST-2         144.156               0</span></span>
<span><span class="co">#&gt; SAMEA103885043             TEST-3          71.239               0</span></span>
<span><span class="co">#&gt; SAMEA103885392             TEST-4         130.372               0</span></span>
<span><span class="co">#&gt; SAMEA103885182             TEST-5         241.527               0</span></span>
<span><span class="co">#&gt; SAMEA103885136             TEST-6         140.124               0</span></span>
<span><span class="co">#&gt;                ENSG00000000419 ENSG00000000457 ENSG00000000460 ENSG00000000938</span></span>
<span><span class="co">#&gt; SAMEA103885102        1462.001         616.732         913.913         210.999</span></span>
<span><span class="co">#&gt; SAMEA103885347        1380.000        1058.364         330.360         273.001</span></span>
<span><span class="co">#&gt; SAMEA103885043        1117.000         436.315         292.047         635.000</span></span>
<span><span class="co">#&gt; SAMEA103885392         961.000         703.578         119.337        1210.000</span></span>
<span><span class="co">#&gt; SAMEA103885182        1399.000         580.188         492.361         528.000</span></span>
<span><span class="co">#&gt; SAMEA103885136        1276.000        1036.716         173.410         669.000</span></span>
<span><span class="co">#&gt;                ENSG00000000971 ENSG00000001036 ENSG00000001084</span></span>
<span><span class="co">#&gt; SAMEA103885102         508.000        7715.294        6363.152</span></span>
<span><span class="co">#&gt; SAMEA103885347        8355.997        6376.500       15601.710</span></span>
<span><span class="co">#&gt; SAMEA103885043         657.999        6231.000        2356.371</span></span>
<span><span class="co">#&gt; SAMEA103885392        6407.001        5307.074        8890.247</span></span>
<span><span class="co">#&gt; SAMEA103885182         140.999        5099.999        5275.775</span></span>
<span><span class="co">#&gt; SAMEA103885136        6670.997        4314.000       14894.069</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="converting-metabolon-client-data-table-to-summarizedexperiment">Converting Metabolon client data table to SummarizedExperiment<a class="anchor" aria-label="anchor" href="#converting-metabolon-client-data-table-to-summarizedexperiment"></a>
</h3>
<p>A central functionality is the ability to convert Metabolon Client
data table files to SummarizedExperiment objects. You can do this using
the function <code>cdt_to_se</code>, the output will be a
SummarizedExperiment object that you can use in your analysis pipelines.
It automatically merges the compound information and the subject metdata
and adds them respectively to the rowData and colData of The
SummarizedExperiment object.</p>
<p>You can specify which annotation type you want to use for the
rownames of the SummarizedExperiment object using the argument
<code>rowdata_key</code>. Note: the compounds that do not have the
specified annotation type will be removed from the final object, as the
rownames cannot be duplicated or empty.</p>
<p>The <code>data_type</code> argument allows you to select which type
of data you want to load from the client data table.</p>
<p>You can also save the output SummarizedExperiment object as an RDS
file using the <code>output_file</code> argument. If it is set to
<code>FALSE</code>, the object is just returned without saving, the
output_file argument in this case is of course ignored.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cdt_to_se.html">cdt_to_se</a></span><span class="op">(</span></span>
<span>  cdt <span class="op">=</span> <span class="va">cdt_path</span>,</span>
<span>  output_file <span class="op">=</span> <span class="st">"test/cdt_to_se_bni.rds"</span>,</span>
<span>  save_file <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  data_type <span class="op">=</span> <span class="st">"batch_norm_imputed"</span>,</span>
<span>  rowdata_key <span class="op">=</span> <span class="st">"INCHIKEY"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">se</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 6 24 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(6): FAKEINCHIKEY00001 FAKEINCHIKEY00002 ... FAKEINCHIKEY00005</span></span>
<span><span class="co">#&gt;   FAKEINCHIKEY00006</span></span>
<span><span class="co">#&gt; rowData names(21): CHEM_ID LIB_ID ... PUBCHEM PLATFORM</span></span>
<span><span class="co">#&gt; colnames(24): TEST-1 TEST-2 ... TEST-23 TEST-24</span></span>
<span><span class="co">#&gt; colData names(34): PARENT_SAMPLE_NAME CLIENT_IDENTIFIER ... TREATMENT</span></span>
<span><span class="co">#&gt;   WT_KO</span></span></code></pre></div>
<p>A small trick: you can convert the SummarizedExperiment object to a
SingleCellExperiment object to be able to store additional information
such as dimensionality reductions in it.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'SingleCellExperiment'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:edgeR':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     cpm, tpm</span></span>
<span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"SingleCellExperiment"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># As you can see here we can add dimensionality reductions (as many as we need) to the object using the capabilities of a single cell summarized experiment</span></span>
<span><span class="va">pca_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">sce</span>, <span class="st">"PCA"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">pca_result</span><span class="op">$</span><span class="va">x</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This document has been generated with…</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.2 (2025-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] SingleCellExperiment_1.32.0 DESeq2_1.50.2              </span></span>
<span><span class="co">#&gt;  [3] edgeR_4.8.0                 limma_3.66.0               </span></span>
<span><span class="co">#&gt;  [5] SummarizedExperiment_1.40.0 Biobase_2.70.0             </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.62.0        Seqinfo_1.0.0              </span></span>
<span><span class="co">#&gt;  [9] IRanges_2.44.0              S4Vectors_0.48.0           </span></span>
<span><span class="co">#&gt; [11] BiocGenerics_0.56.0         generics_0.1.4             </span></span>
<span><span class="co">#&gt; [13] MatrixGenerics_1.22.0       matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt; [15] macrophage_1.26.0           metabolonIO_0.2.0          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] gtable_0.3.6        xfun_0.54           bslib_0.9.0        </span></span>
<span><span class="co">#&gt;  [4] ggplot2_4.0.1       lattice_0.22-7      vctrs_0.6.5        </span></span>
<span><span class="co">#&gt;  [7] tools_4.5.2         parallel_4.5.2      tibble_3.3.0       </span></span>
<span><span class="co">#&gt; [10] pkgconfig_2.0.3     Matrix_1.7-4        RColorBrewer_1.1-3 </span></span>
<span><span class="co">#&gt; [13] S7_0.2.1            desc_1.4.3          readxl_1.4.5       </span></span>
<span><span class="co">#&gt; [16] lifecycle_1.0.4     compiler_4.5.2      farver_2.1.2       </span></span>
<span><span class="co">#&gt; [19] textshaping_1.0.4   statmod_1.5.1       codetools_0.2-20   </span></span>
<span><span class="co">#&gt; [22] htmltools_0.5.9     sass_0.4.10         yaml_2.3.11        </span></span>
<span><span class="co">#&gt; [25] pkgdown_2.2.0       pillar_1.11.1       jquerylib_0.1.4    </span></span>
<span><span class="co">#&gt; [28] BiocParallel_1.44.0 DelayedArray_0.36.0 cachem_1.1.0       </span></span>
<span><span class="co">#&gt; [31] abind_1.4-8         locfit_1.5-9.12     digest_0.6.39      </span></span>
<span><span class="co">#&gt; [34] fastmap_1.2.0       grid_4.5.2          cli_3.6.5          </span></span>
<span><span class="co">#&gt; [37] SparseArray_1.10.4  magrittr_2.0.4      S4Arrays_1.10.1    </span></span>
<span><span class="co">#&gt; [40] scales_1.4.0        rmarkdown_2.30      XVector_0.50.0     </span></span>
<span><span class="co">#&gt; [43] cellranger_1.1.0    ragg_1.5.0          evaluate_1.0.5     </span></span>
<span><span class="co">#&gt; [46] knitr_1.50          rlang_1.1.6         Rcpp_1.1.0         </span></span>
<span><span class="co">#&gt; [49] glue_1.8.0          jsonlite_2.0.0      R6_2.6.1           </span></span>
<span><span class="co">#&gt; [52] systemfonts_1.3.1   fs_1.6.6</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Edoardo Filippi.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
